### PR TITLE
fuzzing input in result, resolves #76

### LINKED ISSUE #76: Add the "fuzz word present in response" value in the response

### PR DESCRIPTION
**Pull request by delic**

### CODE REVIEW

File: pkg/filter/regex.go [hunk 1]
```diff
@@ -15,12 +15,12 @@ type RegexpFilter struct {
...
 	re, err := regexp.Compile(value)
 	if err != nil {
-		return &RegexpFilter{}, fmt.Errorf("Size filter or matcher (-fs / -ms): invalid value: %s", value)
+		return &RegexpFilter{}, fmt.Errorf("Regexp filter or matcher (-fr / -mr): invalid value: %s", value)
 	}
 	return &RegexpFilter{Value: re, valueRaw: value}, nil
 }
 
-func (f *RegexpFilter) Filter(response *ffuf.Response) (bool, error) {
+func (f *RegexpFilter) Filter(response *ffuf.Response, inputs map[string][]byte) (bool, error) {
```
Comment (joohoi) [inline pkg/filter/regex.go]: You can access the fuzz keywords through `response.Request.Input` and can remove the code passing the input data through the application.

File: pkg/filter/regex.go [hunk 2]
```diff
@@ -15,12 +15,12 @@ type RegexpFilter struct {
 func NewRegexpFilter(value string) (ffuf.FilterProvider, error) {
 	re, err := regexp.Compile(value)
 	if err != nil {
-		return &RegexpFilter{}, fmt.Errorf("Size filter or matcher (-fs / -ms): invalid value: %s", value)
+		return &RegexpFilter{}, fmt.Errorf("Regexp filter or matcher (-fr / -mr): invalid value: %s", value)
```
Comment (joohoi) [inline pkg/filter/regex.go]: Nice catch!

File: main.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func main() {
 	flag.StringVar(&opts.delay, "p", "", "Seconds of `delay` between requests, or a range of random delay. For example \"0.1\" or \"0.1-2.0\"")
 	flag.StringVar(&opts.filterStatus, "fc", "", "Filter HTTP status codes from response. Comma separated list of codes and ranges")
 	flag.StringVar(&opts.filterSize, "fs", "", "Filter HTTP response size. Comma separated list of sizes and ranges")
-	flag.StringVar(&opts.filterRegexp, "fr", "", "Filter regexp")
+	flag.StringVar(&opts.filterRegexp, "fr", "", "Filter regexp, also fuzz keyword allowed")
```
Comment (joohoi) [inline main.go]: I don't think this needs a mention in the help text, as I think the default assumption here is that everything works with the keywords ;)

File: README.md [hunk 4]
```diff
@@ -146,7 +146,7 @@ Usage of ./ffuf:
   -mode string
     	Multi-wordlist operation mode. Available modes: clusterbomb, pitchfork (default "clusterbomb")
   -mr string
-    	Match regexp
+    	Match regexp, also fuzz keyword allowed
```
Comment (joohoi) [inline README.md]: I don't think this needs a mention in the help text, as I think the default assumption here is that everything works with the keywords ;)

File: main.go [hunk 5]
```diff
@@ -86,7 +86,7 @@ func main() {
 	flag.Var(&opts.cookies, "cookie", "Cookie data (alias of -b)")
 	flag.StringVar(&opts.matcherStatus, "mc", "200,204,301,302,307,401,403", "Match HTTP status codes from respose, use \"all\" to match every response code.")
 	flag.StringVar(&opts.matcherSize, "ms", "", "Match HTTP response size")
-	flag.StringVar(&opts.matcherRegexp, "mr", "", "Match regexp")
+	flag.StringVar(&opts.matcherRegexp, "mr", "", "Match regexp, also fuzz keyword allowed")
```
Comment (joohoi) [inline main.go]: I don't think this needs a mention in the help text, as I think the default assumption here is that everything works with the keywords ;)

File: pkg/filter/regex.go [hunk 6]
```diff
@@ -29,6 +29,14 @@ func (f *RegexpFilter) Filter(response *ffuf.Response) (bool, error) {
 	}
 	matchdata := []byte(matchheaders)
 	matchdata = append(matchdata, response.Data...)
+	// is the raw value a fuzzing keyword
+	if _, fuzzed := inputs[f.valueRaw]; fuzzed {
+		matched, err := regexp.Match(string(inputs[f.valueRaw]), matchdata)
```
Comment (joohoi) [inline pkg/filter/regex.go]: We need to use `regexp.QuoteMeta` for the input data in order to properly escape all the regex control characters as the input data might have them.

Comment (delic) [reply pkg/filter/regex.go]: Tried searching such a method with no luck, good to know!

File: pkg/filter/regex.go (hunk 1 above)
Comment (delic) [reply pkg/filter/regex.go]: Ah, missed that, thx.

File: main.go (hunk 3 above)
Comment (delic) [reply main.go]: Removed.

File: README.md (hunk 4 above)
Comment (delic) [reply README.md]: Removed.

File: main.go (hunk 5 above)
Comment (delic) [reply main.go]: Removed.

File: pkg/filter/regex.go (hunk 6 above)
Comment (delic) [reply pkg/filter/regex.go]: It is now fixed.

**Comment by delic** [comment]
Will implement the more complex matching.

**Comment by delic** [comment]
Now for example -mr 'FUZZ *[a-f0-9]+ FUZ2 t' using multiple wordlists works.

**Pull request merged**